ecl2df/summary.py

```diff
@@ -435,6 +431,55 @@ def df(
         )
         logger.warning("Duplicates: %s", list(dframe.columns[dupes]))
         dframe = dframe.loc[:, ~dframe.columns.duplicated()]
+
+    dframe = _ensure_unique_datetime_index(dframe)
+
+    if datetime is True:
+        if dframe.index.dtype == "object":
+            dframe.index = pd.to_datetime(dframe.index)
+
+    return dframe
+
+
+def _ensure_unique_datetime_index(dframe: pd.DataFrame) -> pd.DataFrame:
+    index_duplicates = dframe.index.duplicated(keep="first")
+    if any(index_duplicates):
+        index_duplicate_log_string = ""
+        for idx in dframe.index[index_duplicates]:
+            index_duplicate_log_string += f"\n{idx}"
+
+        if "TIMESTEP" in dframe:
+            logger.info(
+                "Dataframe of smry data contained duplicate timestamps due to limited",
+                " output resolution. Vector TIMESTEP exists, utilizing it to create "
+                "discrete timestamps",
+                f"Original duplicates were:{index_duplicate_log_string}",
+            )
+            index_as_list = dframe.index.to_list()
+
+            if dframe.attrs["meta"]["TIMESTEP"]["unit"] == "DAYS":
```

Comment:
I would guess you get a "KeyError" here if the unit is not available, so you will probably not go to the ValueError below. (?)

Reply:
The idea was to catch a scenario where unit exists, but not equal to DAYS or HOURS. It was not to catch a scenario where the unit key was missing. Think the existence of the unit key is pretty much ensured by running the smry_meta command and adding it to the dataframe (currently lines 418-422)